### PR TITLE
Fix getKeyHash utility for the Guid values (T721610)

### PR DESCRIPTION
**Pull request by DokaRus**

### CODE REVIEW

**Review by mrjohn0011 — APPROVED** [review]
Split ES6 conversion and the fix into 2 different commits. It is hard to review summary changes

File: js/core/utils/common.js [hunk 1]
```diff
@@ -268,10 +266,12 @@ var equalByValue = function(object1, object2, deep) {
     return false;
 };
 
-var getKeyHash = function(key) {
-    if(typeUtils.isObject(key) || Array.isArray(key)) {
+const getKeyHash = function(key) {
+    if(key instanceof Guid) {
```
Comment (San4es) [inline js/core/utils/common.js:270]: Can we use valueOf here?
```
    if (isFunction(key.valueOf)) {
      key= key.valueOf();
    }
```

Comment (DokaRus) [reply js/core/utils/common.js:270]: `Object` and `Array` has the `valueOf` method too at it returns an object. It can break our selection logic

**Review by San4es — APPROVED** [review]
LGTM

**Pull request merged**